### PR TITLE
Allow healthz to prevent new connections before shutdown

### PR DESCRIPTION
**Pull request by mat285**

### CODE REVIEW

File: web/app.go [hunk 1]
```diff
@@ -502,6 +523,17 @@ func (a *App) Start() (err error) {
 	keepAliveListener := TCPKeepAliveListener{a.listener}
 	var shutdownErr error
 	a.latch.Started()
+
+	// start the child app
+	if a.child != nil {
+		go func() {
+			err = a.child.Start()
```
Comment (wcharczuk) [inline web/app.go]: assigning `err` across goroutine boundaries is maybe not best practice, can you use a channel

Comment (mat285) [reply web/app.go]: yep will do

File: web/healthz.go [hunk 2]
```diff
@@ -125,6 +130,20 @@ func (hz *Healthz) DefaultHeaders() map[string]string {
 	return hz.defaultHeaders
 }
 
+// Ready returns if healthz server is available ignoring the underlying server
+func (hz *Healthz) Ready() bool {
+	hz.readyLock.Lock()
```
Comment (wcharczuk) [inline web/healthz.go]: this is what the latches are for

Comment (mat285) [reply web/healthz.go]: 👍

**Comment by wcharczuk** [comment]
It's a good start, but overall inverts the spirit of `healthz` which is as an add-on to an app to make it more production ready (where the app is just the stuff the user needs to build, and then slaps healthz on it)

**Comment by wcharczuk** [comment]
sorry didn't mean to close this

**Pull request merged**